app/components/NotificationModal/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, ReactElement } from "react";
+import { useSelector } from "react-redux";
+import { notificationStatus } from "state/selectors";
+import styles from "./index.module.css";
+import AlarmIcon from "@mui/icons-material/Alarm";
+import CloseIcon from "@mui/icons-material/Close";
+import CheckIcon from "@mui/icons-material/Check";
+import ErrorOutlineOutlinedIcon from "@mui/icons-material/ErrorOutlineOutlined";
+import { AppNotificationStatus } from "../../state/app";
+import { setAppState } from "state/app";
+import { useDispatch } from "react-redux";
+import { getStatusMessage, TxnStatus } from "actions/utils";
+import { ClaimExceededModal } from "components/views/PKlima/ClaimExceededModal";
+
+type StatusTypes =
+  | "done"
+  | "error"
+  | "userConfirmation"
+  | "networkConfirmation"
+  | "claimExceeded";
+
+interface Props {}
+
+interface ModalAssetTypes {
+  [key: string]: {
+    [key: string]: string | ReactElement;
+  };
+}
+
+const modalAssets: ModalAssetTypes = {
+  header: {
+    done: "SUCCESS!",
+    error: "FAILURE!",
+    userConfirmation: "CONFIRMATION",
+    networkConfirmation: "PENDING",
+  },
+  iconStyle: {
+    done: styles.icon_success,
+    error: styles.icon_failure,
+    userConfirmation: styles.icon_confirmation,
+    networkConfirmation: styles.icon_confirmation,
+  },
+  iconComponent: {
+    done: <CheckIcon />,
+    error: <ErrorOutlineOutlinedIcon />,
+    userConfirmation: <AlarmIcon />,
+    networkConfirmation: <AlarmIcon />,
+  },
+};
+
+export const NotificationModal: FC<Props> = () => {
+  const dispatch = useDispatch();
+  const status: AppNotificationStatus | null = useSelector(notificationStatus);
+  if (!status) return null;
+  const { statusType, message } = status;
+  if (status && statusType === "claimExceeded") return <ClaimExceededModal />;
+
+  const getAsset = (
+    section: string,
+    status: AppNotificationStatus
+  ): string | ReactElement => {
+    return status.statusType ? modalAssets[section][status.statusType] : "";
+  };
+
+  const closeModal = function (): void {
+    dispatch(setAppState({ notificationStatus: null }));
+  };
+
+  return (
+    <>
+      <div className={styles.bg}>
+        <div className={styles.card}>
+          <div className={styles.card_header}>
+            <p>{getAsset("header", status)}</p>
+            <button onClick={closeModal} className={styles.closeButton}>
```

Comment:
Hm the experience is a bit funky when the user closes before the txn is done. I guess its not a dealbreaker.

How do you feel about disabling this button until the final state is reached? success/error

Reply:
I'm not a UX designer but I like the idea.  I agree the experience is a little weird if you're like me and always click away every popup as soon as I can.  Plus this way it sort of traps the user so metamask is the only thing they can interact with while the modal is in pending mode and can't accidentally go purchase the same bond 4 times if they don't see the pending status in their metamask.

Reply:
done.